src/ngx-smart-modal/src/components/ngx-smart-modal.component.ts

```diff
@@ -282,6 +303,23 @@ export class NgxSmartModalComponent implements OnInit, OnDestroy {
     }
   }
 
+  /**
+   * While modal is open, the focus stay on it
+   */
+  public trapFocusModal = (event: KeyboardEvent) => {
+    if (event.keyCode === 9) {
```

Comment:
Try to refactor it this way to avoid over-cascading :

```typescript
if (
    event.keyCode === 9 && 
    this.nsmDialog && 
    this.nsmContent && 
    this._ngxSmartModalService.getTopOpenedModal() === this &&
    !this.nsmDialog.nativeElement.contains(document.activeElement)
  ) {
      event.preventDefault();
      this.nsmDialog.nativeElement.focus();
    }
```